docs/kb/integrations/customize-semgrep-precommit.md

```diff
@@ -0,0 +1,87 @@
+---
+description: Understand how to customize Semgrep's behavior when using it with pre-commit.
+tags:
+  - Extensions
+---
+
+# Customize Semgrep in pre-commit
+
+Semgrep offers two standard pre-commit hooks [outlined here](/docs/extensions/overview/#pre-commit).
+
+* The `semgrep` hook is optimized to run with a provided set of rules, either local rules or rules from the Semgrep Registry.
+* The `semgrep-ci` hook is optimized to run with the rules from your organization in the Semgrep AppSec Platform, without sending results to the platform, since results from pre-commit are temporary and not linked to the final code in the repository.
+  * This hook requires the user be logged in locally to fetch the rules from the organization.
+
+You can also create your own Semgrep hooks for pre-commit if you have particular needs or preferences, such as if you want users to see the output of the Semgrep scan even if it passed, or if you want to limit the scan to a particular Semgrep product.
+
+## Show the scan output
+
+By default, pre-commit does not show the scan output if the hook passes. If you want your developers to see the output even if non-blocking findings are found, you can set the `verbose` option to print the output. To limit output to findings only (no scan information or diagnostics), redirect `stderr` to `/dev/null`, or use `--quiet`.
+
+This example is based on the `semgrep-ci` hook, but a similar adjustment would also work with the `semgrep` hook.
+
+```yaml
+repos:
+- repo: https://github.com/semgrep/pre-commit
+  rev: 'v1.101.0'
+  hooks:
+    - id: semgrep-verbose
+      entry: semgrep
+      args: ["ci", "--dry-run", "--baseline-commit", "HEAD" "2>/dev/null"]
+      verbose: true
+      pass_filenames: false
+```
+
+## Limit the scan to a particular product
+
+Semgrep Secrets is an ideal product to run before commit, since it can help prevent secrets from ever making it into the Git history, even locally. To run only Secrets in pre-commit, add the product flag to the `args`:
+
+```yaml
+- repo: https://github.com/semgrep/pre-commit
+  rev: 'v1.101.0'
+  hooks:
+    - id:  semgrep-secrets
+      pass_filenames: false
+      args: ["ci", "--dry-run", "--baseline-commit", "HEAD", "--secrets"]
+```
+
+## Scan with Pro Engine
+
+The `semgrep-ci` hook requires the user to be logged in locally and runs with the engine configured in the organization, but the standard `semgrep` hook can also take advantage of local login to run with the Pro Engine.
```

Comment:
```suggestion
The `semgrep-ci` hook requires the user to be logged in locally and runs with the engine configured in the organization, but the standard `semgrep` hook can also take advantage of local login to run with interfile analysis.
```

Reply:
also wrong for the same reason, sadly. I don't have a ready suggestion here.

Reply:
```suggestion
The `semgrep-ci` hook requires the user to be logged in locally and runs with the engine configured in the organization, but the standard `semgrep` hook can also take advantage of local login to run with Pro rules and cross-function analysis.
```
Does this work?

Reply:
yeah, I think that's fine and probably clearer than Semgrep Code! Thanks!